timelock-server-benchmark-client/src/main/java/com/palantir/atlasdb/timelock/benchmarks/benchmarks/AbstractRangeScanBenchmark.java

```diff
@@ -0,0 +1,180 @@
+        if (existingBucket.isPresent()) {
+            this.bucket = existingBucket.get();
+            log.info("found existing test data under bucket {}", bucket);
+            return;
+        }
+
+        this.bucket = UUID.randomUUID().toString();
+        log.info("creating new test data under bucket {}", bucket);
+
+        writeData();
+
+        recordMetadata();
+    }
+
+    private void writeData() {
+        int numWritten = 0;
+        AtomicLong key = new AtomicLong(0);
+        byte[] data = RandomBytes.ofLength(dataSize);
+
+        while (numWritten < numRows) {
+            int numToWrite = Math.min(numRows - numWritten, batchSize);
+
+            txnManager.runTaskWithRetry(txn -> {
+                Map<Long, byte[]> values = Maps.newHashMap();
+                for (int i = 0; i < numToWrite; i++) {
+                    values.put(key.getAndIncrement(), data);
+                }
+
+                writeValues(txn, values);
+                return null;
+            });
+
+            numWritten += numToWrite;
+        }
+    }
+
+    @Override
+    protected final void performOneCall() {
+        List<byte[]> results = txnManager.runTaskReadOnly(txn -> getRange(txn, 0L, numRows));
+
+        Preconditions.checkState(results.size() == numRows);
+        for (byte[] resultData : results) {
+            Preconditions.checkState(resultData.length == dataSize);
+        }
+    }
+
+    protected abstract List<byte[]> getRange(Transaction txn, long startInclusive, long endExclusive);
+
+    protected abstract void writeValues(Transaction txn, Map<Long, byte[]> valuesByKey);
+
+    @Override
+    public Map<String, Object> getExtraParameters() {
+        return ImmutableMap.of("numRows", numRows, "dataSize", dataSize);
+    }
+
+    public static void main(String[] args) {
+        Metadata metadata = new Metadata("foo");
+        byte[] data = serialize(metadata);
+        deserialize(data);
+    }
+
+    private void recordMetadata() {
+        txnManager.runTaskWithRetry(txn -> {
+            MetadataTable table = BenchmarksTableFactory.of().getMetadataTable(txn);
+
+            Metadata metadata = new Metadata(bucket);
+            table.putData(MetadataTable.MetadataRow.of(getKeyForParameters()), serialize(metadata));
+            return null;
+        });
+    }
+
+    private Optional<String> getExistingBucketForParameters() {
+        return txnManager.runTaskReadOnly(txn -> {
+            MetadataTable table = BenchmarksTableFactory.of().getMetadataTable(txn);
+
+            return table.getRow(MetadataTable.MetadataRow.of(getKeyForParameters()))
+                    .map(MetadataTable.MetadataRowResult::getData)
+                    .map(AbstractRangeScanBenchmark::deserialize)
+                    .map(mdata -> mdata.bucket);
+        });
+    }
+
+    private static byte[] serialize(Metadata metadata) {
+        try {
+            return new ObjectMapper().writeValueAsBytes(metadata);
```

Comment:
we should pull the ObjectMapper out as a `static final` and use Afterburner, to avoid thrashing and testing too much serDe

```
private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new AfterburnerModule());
```